src/Scouting.elm

```diff
@@ -1,141 +1,198 @@
-            }
-
-
-view : Model -> Html.Html Msg
-view model =
-    Html.div []
-        [ Html.text "Pre-Scout screen:"
-        , Html.div []
-            [ Html.input
-                [ placeholder "Scouter's name..."
-                , onInput NameInput
-                , value model.scouterName
-                ]
-                []
-            , Html.input
-                [ placeholder "Team's number..."
-                , onInput TeamInput
-                , value <|
-                    if 0 == model.teamNum then
-                        ""
-
-                    else
-                        String.fromInt model.teamNum
-                ]
-                []
-            , Html.input
-                [ placeholder "Match number..."
-                , onInput MatchInput
-                , value <|
-                    if 0 == model.matchNum then
-                        ""
-
-                    else
-                        String.fromInt model.matchNum
-                ]
-                []
-            , Html.input
-                [ style "visibility" dropDown_driverStation
-                , placeholder "Driver station..."
-                , onInput DriverStationInput
-                , value model.driverStation
-                ]
-                []
-            , Html.button [ onClick Start ]
-                [ Html.text <|
-                    if model.isStarted then
-                        "Started"
-
-                    else
-                        "Startn't"
-                ]
-            ]
-        , Html.div []
-            [ Html.text <|
-                "Status:  name - "
-                    ++ model.scouterName
-                    ++ ", team - "
-                    ++ String.fromInt model.teamNum
-                    ++ ", match - "
-                    ++ String.fromInt model.matchNum
-                    ++ ", station - "
-                    ++ model.driverStation
-            ]
-        ]
+module Scouting exposing (Model, Msg, driverStations, init, main, update, view)
+
+import Browser
+import Html
+import Html.Attributes exposing (placeholder, style, value)
+import Html.Events exposing (onClick, onInput)
+import List.Extra exposing (elemIndex, getAt)
+import Maybe exposing (withDefault)
+import Maybe.Extra exposing (unwrap)
+import Tuple exposing (first, second)
+
+
+main : Program () Model Msg
+main =
+    Browser.sandbox
+        { init = init
+        , view = view
+        , update = update
+        }
+
+
+type Msg
+    = NameInput String
+    | TeamInput String
+    | MatchInput String
+    | Start
+
+
+type alias Model =
+    { scouter : String
+    , team : Maybe Int
+    , match : Maybe Int
+    , driverStation : String
+    , isStarted : Bool
+    }
+
+
+driverStations : List (List Int)
+driverStations =
+    [ [ 1690, 1574, 3339, 254, 2056, 1323 ]
+    , [ 118, 1577, 1024, 2056, 1690, 254 ]
+    , [ 1574, 3339, 1577, 1323, 1024, 118 ]
+    , [ 3339, 1574, 1577, 1024, 118, 2056 ]
+    ]

```

Comment:
This doesn't belong here - define such dummy data in `Main.elm` or even in a dedicated `Dummy.elm` module.

Reply:
Also, are you sure this is the best way to model such data?
Isn't a record better?
If you have something like:
```elm
type Positive number =
    Positive number


makePositive : number -> Maybe (Positive number)
makePositive n =
    if n > 0 then
        Positive n

    else
        Nothing


type alias TeamNumber =
    Positive Int


type alias MatchNumber =
    Positive Int


type alias Alliance =
    { one : TeamNumber
    , two : TeamNumber
    , three : TeamNumber
    }


type alias Match =
    { number : MatchNumber
    , red : Alliance
    , blue : Alliance
    }
```
With all these, you can't make negative team or match numbers, can't have an alliance with less than or more than 3 teams, and a match without a number and two alliances. Therefore, this design just eliminates all these kinds of bugs.

Reply:
This is the thing that I failed to do in the last commit-
I just didn't manage to extract the driver station from there given the team and match numbers.
Is there an easy way to do this with records?
And what advantages does it have over a list of lists, when I only read from it and not write to it?

Reply:
I don't quite follow - what's not easy about https://github.com/orbit1690/Scouting-2019/pull/2#discussion_r353376318?
Records are just one kind of types - a product type. Using both products and sums (where appropriate) can result in a much more coherent (and safe) design.
You should read https://guide.elm-lang.org/appendix/types_as_sets.html for a thorough overview on this matter.

To your question - the suggested design has the advantage of making almost all invalid states unrepresentable - you can't make an `Alliance` of less than or more than 3 teams, your teams are guaranteed to be valid (no non-positive team numbers), and a `Match` must have exactly two `Alliance`s and a `MatchNumber` (which also must be positive).

Reply:
> Can you be more specific?

I didn't manage to get the 'driverStation' from the 'matches' function in Match.elm-
tho I succeeded already.

> You should read https://guide.elm-lang.org/appendix/types_as_sets.html for a thorough overview on this matter.

👍 

>  you can't make an Alliance of less than or more than 3 teams, your teams are guaranteed to be valid (no non-positive team numbers), and a Match must have exactly two Alliances and a MatchNumber (which also must be positive).

I understand the benefit of not being able to make alliances or matches of too few / many teams, but the teams already cannot be entered as negative numbers: the input does not allow me to write things other than numbers, so '-' is not allowed.